src/networking.c

```diff
@@ -4292,6 +4292,34 @@ int stopThreadedIOIfNeeded(void) {
     }
 }
 
+/* This function increases the number of threads to server.io_threads_num.
+ * It returns C_OK when the update is successful, otherwise it returns
+ * C_ERR when trying to decrease the number of threads. */
+int updateThreadedIO(void) {
+    /* Return when trying to decrease the number of threads */
+    if (server.io_threads_num > server.config_io_threads_num) return C_ERR;
+
+    /* Create the list of clients for the main thread if not created already */
+    if (!io_threads_list[0]) io_threads_list[0] = listCreate();
+
+    /* Create new threads */
+    for (int i = server.io_threads_num; i < server.config_io_threads_num; i++) {
+        io_threads_list[i] = listCreate();
+        pthread_t tid;
+        pthread_mutex_init(&io_threads_mutex[i], NULL);
+        setIOPendingCount(i, 0);
+        if (!server.io_threads_active)
+            pthread_mutex_lock(&io_threads_mutex[i]);
+        if (pthread_create(&tid, NULL, IOThreadMain, (void*)(long)i) != 0) {
+            serverLog(LL_WARNING, "Fatal: Can't initialize IO thread.");
+            exit(1);
```

Comment:
we shoule exit if we can't create the io thread?

Reply:
If the thread fails to create I don't think it's helpful to keep going, if we don't have resources it may cause problems in other parts of the code. Maybe we could use the `serverPanic` function to exit in a cleaner way?